js/pages/settings.js

```diff
@@ -428,15 +441,81 @@ const loadSignInInformationElements = () => {
 };
 
 const handleEditSignInInformationSection = () => {
-  btnObj.changeLoginButton.addEventListener('click', () => {
+  btnObj.changeLoginButton.addEventListener('click', async () => {
+    
     successMessageElement = hideSuccessMessage(successMessageElement);
-    formVisBools.isLoginFormDisplayed = toggleElementVisibility(loginElementArray, formVisBools.isLoginFormDisplayed);
     if (formVisBools.isLoginFormDisplayed) {
-      hideOptionalElementsOnShowForm([optRowEles.loginEmailRow, optRowEles.loginPhoneRow]);
-      toggleActiveForm(FormTypes.LOGIN);
-      openUpdateLoginForm({ currentTarget: document.getElementsByClassName('tablinks')[0] }, 'form1');
+        //If the form is displayed then this is acting as the cancel button
+        formVisBools.isLoginFormDisplayed = toggleElementVisibility(loginElementArray, formVisBools.isLoginFormDisplayed);
+        toggleButtonText();
+    } else {
+        //Display the reauth dialog
+        const reauthModal = document.getElementById('reauthModal');
+    
+        const signInBtn = reauthModal.querySelector('#signInBtn');
+        const closeBtn = reauthModal.querySelector('#reauthClose');
+        const headerClose = reauthModal.querySelector('.close');
+        const accountInput = reauthModal.querySelector('#accountInput');
+    
+        let usGov = '';
+    
+        const handleSignInBtn = async (e) => {
+            e.preventDefault();
+            window.localStorage.setItem('signInUpdate', 'yes');
+            const inputStr = accountInput.value.trim();
+            const isEmail = !!inputStr.match(validEmailFormat);
+            const isPhone = !!inputStr.match(validPhoneNumberFormat);
+        
+            const ui = firebaseui.auth.AuthUI.getInstance() || new firebaseui.auth.AuthUI(firebase.auth());
+            if (isEmail) {
+            //   await signInAnonymously();
+              const emailForQuery = inputStr
+                .replaceAll('%', '%25')
+                .replaceAll('#', '%23')
+                .replaceAll('&', '%26')
+                .replaceAll(`'`, '%27')
+                .replaceAll('+', '%2B');
+        
+              const response = await checkAccount({
+                accountType: 'email',
+                accountValue: emailForQuery,
+              });
+        
+              if (response?.data?.accountExists) {
+                const account = { type: 'email', value: inputStr };
+                firebaseSignInRender({ ui, account, usGov, signInConfig, signInConfigDev });
+              } else {
+                alert('Account Not Found');
+              }
+            } else if (isPhone) {
+            //   await signInAnonymously();
```

Comment:
Should we also remove this?

Reply:
I believe @bransteitterbr  has some purpose for that.